plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageSettings.java

```diff
@@ -43,20 +45,43 @@
         Setting.affixKeySetting(Storage.PREFIX, "key", (key) -> Setting.simpleString(key, Setting.Property.NodeScope));
     private static final Setting<Boolean> DEFAULT_SETTING =
         Setting.affixKeySetting(Storage.PREFIX, "default", (key) -> Setting.boolSetting(key, false, Setting.Property.NodeScope));
+    /** The host name of a proxy to connect to s3 through. */
+    private static final Setting<String> PROXY_HOST_SETTING =
+        Setting.affixKeySetting(Storage.PREFIX, "proxy.host", (key) -> Setting.simpleString(key, Setting.Property.NodeScope));
+    /** The port of a proxy to connect to azure through. */
+    private static final Setting<Integer> PROXY_PORT_SETTING = Setting.affixKeySetting(Storage.PREFIX, "proxy.port",
+        key -> Setting.intSetting(key, 80, 0, 1<<16, Setting.Property.NodeScope));
```

Comment:
Maximum value allowed here is 65536 but later in the code `new InetSocketAddress(proxyHost, proxyPort))` checks that _A valid port value is between 0 and 65535_. Max value here should be 65535.